### PR TITLE
Added TypeScript support (d.ts file generation) for Azure.NodeJS, sam…

### PR DESCRIPTION
**Pull request by BretJohnson**
…e as was done for NodeJS

### CODE REVIEW

**Comment by amarzavery** [comment]
Fix fxcop/code analysis errors:

  Running Code Analysis...
d:\workspace\autorest\AutoRest\Generators\NodeJS\NodeJS\ClientModelExtensions.cs(514): error CA1307: Microsoft.Globalization : Because the behavior of 'string.StartsWith(string)' could vary based on the current user's locale settings, replace this call in 'ClientModelExtensions.TSType(this IType, bool)' with a call to 'string.StartsWith(string, StringComparison)'. If the result of 'string.StartsWith(string, StringComparison)' will be displayed to the user, such as when sorting a list of items for display in a list box, specify 'StringComparison.CurrentCulture' or 'StringComparison.CurrentCultureIgnoreCase' as the 'StringComparison' parameter. If comparing case-insensitive identifiers, such as file paths, environment variables, or registry keys and values, specify 'StringComparison.OrdinalIgnoreCase'. Otherwise, if comparing case-sensitive identifiers, specify 'StringComparison.Ordinal'. [D:\workspace\autorest\AutoRest\Generators\NodeJS\NodeJS\AutoRest.Generator.NodeJS.csproj]
d:\workspace\autorest\AutoRest\Generators\NodeJS\NodeJS\ClientModelExtensions.cs(514): error CA1307: Microsoft.Globalization : Because the behavior of 'string.StartsWith(string)' could vary based on the current user's locale settings, replace this call in 'ClientModelExtensions.TSType(this IType, bool)' with a call to 'string.StartsWith(string, StringComparison)'. If the result of 'string.StartsWith(string, StringComparison)' will be displayed to the user, such as when sorting a list of items for display in a list box, specify 'StringComparison.CurrentCulture' or 'StringComparison.CurrentCultureIgnoreCase' as the 'StringComparison' parameter. If comparing case-insensitive identifiers, such as file paths, environment variables, or registry keys and values, specify 'StringComparison.OrdinalIgnoreCase'. Otherwise, if comparing case-sensitive identifiers, specify 'StringComparison.Ordinal'. [D:\workspace\autorest\AutoRest\Generators\NodeJS\NodeJS\AutoRest.Generator.NodeJS.csproj]
  Code Analysis Complete -- 2 error(s), 0 warning(s)
MSBUILD : error : Code Analysis detected errors.  See Code Analysis results window or log file for details. [D:\workspace\autorest\AutoRest\Generators\NodeJS\NodeJS\AutoRest.Generator.NodeJS.csproj]

**Comment by BretJohnson** [comment]
FYI - The Travis CI failure is in C# tests.   That problem looks intermittent (as they passed on the previous commit) and unrelated to any of my changes, which were all in NodeJS/TypeScript related code.

**Comment by amarzavery** [comment]
@BretJohnson - I have posted the error from the Jenkins CI which is on a Windows machine. 
If you right click on the solution in Visual Studioand click Run Code Analysis then you should be able to see the code analysis errors.

**Comment by BretJohnson** [comment]
@amarzavery I fixed the two StartsWith StyleCop warnings (above) earlier.   Let me know if there's anything else.

**Comment by BretJohnson** [comment]
Can you guys merge this.  Thanks.

File: AutoRest/Generators/NodeJS/NodeJS/NodeJSCodeGenerator.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace Microsoft.Rest.Generator.NodeJS
     public class NodeJSCodeGenerator : CodeGenerator
     {
         private const string ClientRuntimePackage = "ms-rest version 1.2.0";
-        private const bool DisableTypeScriptGeneration = false;    // Change to true if you want to no longer generate the 3 d.ts files, for some reason
+        public const bool DisableTypeScriptGeneration = false;    // Change to true if you want to no longer generate the 3 d.ts files, for some reason
```
Comment (stankovski) [inline AutoRest/Generators/NodeJS/NodeJS/NodeJSCodeGenerator.cs]: Should this be exposed as a setting? If so, change this to a property and add an attribute:

```csharp
[SettingsInfo("Disables TypeScript generation.")]
public bool DisableTypeScriptGeneration {get;set;}
```

**Comment by stankovski** [comment]
@BretJohnson please also call `gulp regenerate:expected` from cmd to regenerate test classes

File: AutoRest/Generators/NodeJS/NodeJS/NodeJSCodeGenerator.cs (hunk 1 above)
Comment (BretJohnson) [inline AutoRest/Generators/NodeJS/NodeJS/NodeJSCodeGenerator.cs]: Probably; i'll change it.   How would a setting like that get set then?

**Comment by stankovski** [comment]
Through cmd AutoRest.exe -DisableTypeScriptGeneration

**Comment by BretJohnson** [comment]
@stankovski I made both of the requested changes.   I'm glad to see that specifying the DisableTypeScriptGeneration flag on the command line is so easy;  that was a feature i was wanting--just in case.

**Pull request merged**